tests/unit/test_task_runner.cpp

```diff
@@ -252,6 +253,60 @@ TEST(TaskRunner, TestDelayedTasksOrder)
     STRCMP_EQUAL("bac", str.c_str());
 }
 
+TEST(TaskRunner, TestCancelDelayedTasks)
+{
+    std::string              str;
+    otbr::TaskRunner         taskRunner;
+    otbr::TaskRunner::TaskId tid1, tid2, tid3, tid4, tid5;
+
+    tid1 = taskRunner.Post(std::chrono::milliseconds(10), [&]() { str.push_back('a'); });
+    tid2 = taskRunner.Post(std::chrono::milliseconds(20), [&]() { str.push_back('b'); });
+    tid3 = taskRunner.Post(std::chrono::milliseconds(30), [&]() { str.push_back('c'); });
+    tid4 = taskRunner.Post(std::chrono::milliseconds(40), [&]() { str.push_back('d'); });
+    tid5 = taskRunner.Post(std::chrono::milliseconds(50), [&]() { str.push_back('e'); });
+
+    CHECK(0 < tid1);
+    CHECK(tid1 < tid2);
+    CHECK(tid2 < tid3);
+    CHECK(tid3 < tid4);
+    CHECK(tid4 < tid5);
+
+    taskRunner.Cancel(tid2);
+
+    taskRunner.Post(std::chrono::milliseconds(10), [&]() { taskRunner.Cancel(tid3); });
+    std::thread([&]() {
```

Comment:
nit: join the thread at end of this test case for just making it stable.

Reply:
done